logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/flowcontrol/AppenderMetrics.java

```diff
@@ -0,0 +1,135 @@
+          .register();
+
+  private static final Counter TOTAL_APPEND_TRY_COUNT =
+      Counter.build()
+          .namespace("zeebe")
+          .name("try_to_append_total")
+          .help("Number of tries to append")
+          .labelNames("partition")
+          .register();
+
+  private static final Gauge CURRENT_INFLIGHT =
+      Gauge.build()
+          .namespace("zeebe")
+          .name("backpressure_inflight_append_count")
+          .help("Current number of append inflight")
+          .labelNames("partition")
+          .register();
+
+  private static final Gauge CURRENT_LIMIT =
+      Gauge.build()
+          .namespace("zeebe")
+          .name("backpressure_append_limit")
+          .help("Current limit for number of inflight appends")
+          .labelNames("partition")
+          .register();
+
+  private static final Gauge LAST_COMMITTED_POSITION =
+      Gauge.build()
+          .namespace("zeebe")
+          .name("log_appender_last_committed_position")
+          .help("The last committed position.")
+          .labelNames("partition")
+          .register();
+
+  private static final Gauge LAST_WRITTEN_POSITION =
+      Gauge.build()
+          .namespace("zeebe")
+          .name("log_appender_last_appended_position")
+          .help("The last appended position by the appender.")
+          .labelNames("partition")
+          .register();
+
+  private static final Histogram WRITE_LATENCY =
+      Histogram.build()
+          .namespace("zeebe")
+          .name("log_appender_append_latency")
+          .help("Latency to append an event to the log in seconds")
+          .labelNames("partition")
+          .register();
+  private static final Histogram COMMIT_LATENCY =
+      Histogram.build()
+          .namespace("zeebe")
+          .name("log_appender_commit_latency")
+          .help("Latency to commit an event to the log in seconds")
+          .labelNames("partition")
+          .register();
+
+  private final Counter.Child deferredAppends;
+  private final Counter.Child triedAppends;
+  private final Gauge.Child inflightAppends;
+  private final Gauge.Child inflightLimit;
+  private final Gauge.Child lastCommitted;
+  private final Gauge.Child lastWritten;
+  private final Histogram.Child commitLatency;
+  private final Histogram.Child appendLatency;
+
+  public AppenderMetrics(final int partitionId) {
+    final var partitionLabel = String.valueOf(partitionId);
+    this.deferredAppends = TOTAL_DEFERRED_APPEND_COUNT.labels(partitionLabel);
+    this.triedAppends = TOTAL_APPEND_TRY_COUNT.labels(partitionLabel);
+    this.inflightAppends = CURRENT_INFLIGHT.labels(partitionLabel);
+    this.inflightLimit = CURRENT_LIMIT.labels(partitionLabel);
+    this.lastCommitted = LAST_COMMITTED_POSITION.labels(partitionLabel);
+    this.lastWritten = LAST_WRITTEN_POSITION.labels(partitionLabel);
+    this.commitLatency = COMMIT_LATENCY.labels(partitionLabel);
+    this.appendLatency = WRITE_LATENCY.labels(partitionLabel);
```

Comment:
:+1: Could you share your findings with the team about this issue? Not just our team I guess, also ZPA